test/iterator-test.js

```diff
@@ -35,35 +42,280 @@ make('iterator is seekable', function (db, t, done) {
   })
 })
 
-make('reverse seek in the middle', function (db, t, done) {
+make('iterator is seekable with buffer', function (db, t, done) {
+  var ite = db.iterator()
+  ite.seek(new Buffer('two'))
+  ite.next(function (err, key, value) {
+    t.error(err, 'no error from next()')
+    t.same(key.toString(), 'two', 'key matches')
```

Comment:
i think that should be a separate pull request, to make this one easier to merge